get_hrc.py

```diff
@@ -106,16 +96,25 @@ def format_proton_data(dat, descrs):
     newdat['month'] = [t.month for t in times.datetime]
     newdat['dom'] = [t.day for t in times.datetime]
     newdat['hhmm'] = np.array([f"{t.hour}{t.minute:02}" for t in times.datetime]).astype(int)
-    newdat['p11'] = np.full(len(times), -1.0e5)
 
     hrc_shield = calc_hrc_shield(newdat)
 
     newdat['hrc_shield'] = hrc_shield
 
     hrc_bad = (newdat['p5'] < 0) | (newdat['p7'] < 0) | (newdat['p9'] < 0)
-    newdat['hrc_shield'][hrc_bad] = -1.0e5  # flag bad inputs
+    newdat['hrc_shield'][hrc_bad] = BAD_VALUE  # flag bad inputs
+
+    # Take the Table and make it into an ndarray with the supplied type
+    arr = np.ndarray(len(newdat), dtype=descrs)
```

Comment:
Regarding the suggestion that I initialize this np structured array to start, I'm a little unclear about how one would use np.full to do that.  Would you @taldcroft want me to fill in all the date values with our "channel" bad value of -1e5 to start?  It just feels wrong.  I suppose I could just put the descrs dtype in the file and use np.full_like and assign a fill type for each of the columns.

Reply:
Just seems a bit adding work.

Reply:
This strategy is efficient and seems to work.  But if it feels wrong then the current code will also work.
```
In [4]: nt                                                                      
Out[4]: 
array([(1, 1., 'c'), (2, 2., 'd'), (3, 3., 'e')],
      dtype=[('a', '<i8'), ('b', '<f8'), ('c', '<U1')])

In [6]: out = np.full(3, dtype=nt.dtype, fill_value=-100000)                    

In [7]: out                                                                     
Out[7]: 
array([(-100000, -100000., '-'), (-100000, -100000., '-'),
       (-100000, -100000., '-')],
      dtype=[('a', '<i8'), ('b', '<f8'), ('c', '<U1')])
```

Reply:
Right.  It just felt wrong to me to fill "dom" for example with -1e5.

Reply:
The silly types for the "parts of time" that we seem to have here in this file are not very time-specific.

Reply:
If we are using a convention of -100000 for "missing" or "uninitialized", then `dom` of that is fine. But anyway I just approved this as is.